### ISSUE TITLE
fix watchQuery fetchResults docs

### ISSUE THREAD

**Issue by vytautas-pranskunas-** [zino-app/graphql-flutter#821]
Hi,

I have implemented watchQuery as specified in example but it is not hit:
```
client.watchQuery(
  WatchQueryOptions(
    document: myQuery,
    variables: {},
  ),
)
.stream
.listen((r) {
print(r);
});
;
```

print never triggered

however if i do simple .query everything works fine

**Comment by micimize** [comment]
Looks like the docs are wrong – you need to either pass `fetchResults: true` or call `client.watchQuery().fetchResults()` to trigger the operation

**Comment by vytautas-pranskunas-** [comment]
Then it could be cool if doc is fixed.
Also, correct me if I am wrong - stream should fire twice
1) result from cache
2) result from network with cache update

right?

then why we need to unsubscribe? This stream has an end - so why under the hood it does not do .take(2) ?

**Comment by micimize** [comment]
@vytautas-pranskunas- `watchQuery` has a `pollInterval` option for periodic refetches, as well as a [cache rebroadcasting](https://github.com/zino-app/graphql-flutter/tree/beta/packages/graphql#rebroadcasting) which sends an updated version of the query when underlying data in the cache changes

**Comment by micimize** [comment]
fixed this in #830
